Configure lower NFS grace period.

jsafrane opened the pull request:
e2e tests for NFS is quite slow due to NFS grace period. This patch lowers it to 10 seconds (i.e. the NFS server is actually usable after 10 seconds and not 90 as before).

Also, I rebased the images to CentOS to get better rpc.nfsd and ~40 MB smaller images.

a-robinson commented:
ok to test

thockin commented on contrib/for-tests/volumes-tester/gluster/Dockerfile:
```diff
@@ -12,10 +12,10 @@
 MAINTAINER Jan Safranek, jsafrane@redhat.com
-ENV DEBIAN_FRONTEND noninteractive
-RUN apt-get update -qq && apt-get install -y glusterfs-server -qq
+ADD gluster.repo /etc/yum.repos.d/
```
gluster stuff is unrelated to this PR description - mistake or insufficient description?

thockin replied:
still need to drop these changes?

jsafrane replied:
Having one common base image should save some download on test systems, we don't need to download Fedora *and* Ubuntu, one of them is enough.

thockin replied:
I still don't see what this is in an NFS-related PR - am I missing something?

thockin commented on contrib/for-tests/volumes-tester/gluster/Makefile:
```diff
@@ -1,6 +1,6 @@
 all: push
 
-TAG = 0.1
```
ditto

thockin commented on contrib/for-tests/volumes-tester/nfs/Dockerfile:
```diff
@@ -12,10 +12,9 @@
 MAINTAINER Jan Safranek, jsafrane@redhat.com
-ENV DEBIAN_FRONTEND noninteractive
-RUN apt-get update -qq && apt-get install -y nfs-kernel-server -qq
+RUN yum -y install /usr/bin/ps nfs-utils && yum clean all
```
Why is this re-base needed?  Who is responsible for building and pushing this image?

jsafrane replied:
Sorry, I should have mention it in the log message, rebase to Fedora is needed to have rpc.nfsd with -G option, Ubuntu (14.04) does not compile it in for some reason. I'll update the commit.

Building and pushing was done by @saad-ali, e.g. in #9019.

thockin replied:
Was already done or will be done?

jsafrane replied:
Will be done once these patches are approved and pushed.

jsafrane replied:
Sorry, I got the impression that "LGTM, this is tests-only" was for whole PR. Gluster is now removed.

thockin commented:
NFS parts LGTM.  drop gluster

thockin commented:
LGTM, this is tests-only @quinton-hoole ?  If you can get the shippable and Jenkins passing tomorrow we can consider this.  Else probably not until after 1.0

Shippable says: 

```!!! Error in ./hack/test-integration.sh:49
'"${KUBE_OUTPUT_HOSTBIN}/integration" --v=${LOG_LEVEL} --api-version="$1" --max-concurrency="${KUBE_INTEGRATION_TEST_MAX_CONCURRENCY}"' exited with status 255
Call stack:
1: ./hack/test-integration.sh:49 runTests(...)
2: ./hack/test-integration.sh:63 main(...)
Exiting with status 1
```

jsafrane commented:
Can you share full shippable log? hack/test-integration.sh works for me and I doubt any changes in contrib/ can break it.

thockin commented:
@jsafrane https://gist.github.com/thockin/3155455333f6eb7ee71c

jsafrane commented:
> F0630 07:12:23.741026 32167 integration.go:413] FAILED: pods never started running timed out waiting for the condition

That's [runReplicationControllerTest()](../blob/master/cmd/integration/integration.go#L413), which just creates replication controller from [cmd/integration/v1-controller.json](../blob/master/cmd/integration/v1-controller.json), I don't see how this pull request could break the test. Are you sure it's caused by this pull request? Can't it be random hiccup?

jsafrane commented:
Rebased to current master.

jsafrane commented:
All test passed, can it be merged now?

markturansky commented:
needs rebase.

jsafrane committed:
Configure lower NFS grace period.

From RFC 3530:
  During the grace period, the server must reject READ and WRITE operations
  and non-reclaim locking requests (i.e., other LOCK and OPEN operations)
  with an error of NFS4ERR_GRACE.

That basically means that all open() calls from clients are blocked until the
grace period is over (90 seconds by default).

We want the grace period as low as possible to speed up the tests. '10'
seconds were tested on Fedora 21 and Ubuntu 15.04 as the hosts.

The test image is rebased to Fedora in order to get 'rpc.nfsd -G <n>' option,
Ubuntu does not support it.

thockin commented:
LGTM

thanks, please send gluster in a new PR :)

jsafrane commented:
@saad-ali, can you please upload new container images? In theory:
```
cd test/images/volumes-tester/nfs/
make container
make push
```

Thanks in advance!

saad-ali commented:
> @saad-ali, can you please upload new container images?

Will do

saad-ali commented:
Done
```
6b199664cc26: Image successfully pushed 
Pushing tag for rev [6b199664cc26] on {https://gcr.io/v1/repositories/google_containers/volume-nfs/tags/latest}
Pushing tag for rev [6b199664cc26] on {https://gcr.io/v1/repositories/google_containers/volume-nfs/tags/0.3}
```
